frontend/src/Context/PublisherProvider/usePublisherOptions/usePublisherOptions.spec.tsx

```diff
@@ -42,57 +44,90 @@ const mockUserContextWithCustomSettings = {
 } as UserContextType;
 
 describe('usePublisherOptions', () => {
+  let enumerateDevicesMock: ReturnType<typeof vi.fn>;
+  let deviceManager: DeviceManager;
+  beforeEach(async () => {
+    enumerateDevicesMock = vi.fn();
+    vi.stubGlobal('navigator', {
+      mediaDevices: {
+        enumerateDevices: enumerateDevicesMock,
+      },
+    });
+    Object.defineProperty(window, 'localStorage', {
+      value: localStorageMock,
+      writable: true,
+    });
+    deviceManager = new DeviceManager();
+    enumerateDevicesMock.mockResolvedValue([]);
+    await deviceManager.init();
+  });
+
+  afterAll(() => {
+    window.localStorage.clear();
+  });
+
   it('should use default settings', async () => {
     vi.spyOn(OT, 'hasMediaProcessorSupport').mockReturnValue(true);
     mockUseUserContext.mockImplementation(() => mockUserContextWithDefaultSettings);
     const { result } = renderHook(() => usePublisherOptions());
-    expect(result.current).toEqual({
-      resolution: '1280x720',
-      publishAudio: false,
-      publishVideo: false,
-      audioSource: undefined,
-      videoSource: undefined,
-      insertDefaultUI: false,
-      audioFallback: {
-        publisher: true,
-      },
-      audioFilter: {
-        type: 'advancedNoiseSuppression',
-      },
-      videoFilter: undefined,
-      name: '',
-      initials: '',
+    await waitFor(() => {
```

Comment:
since `usePublisherOptions` have been moved to a `useEffect`, this needed to be `waitFor`'ed 🙂